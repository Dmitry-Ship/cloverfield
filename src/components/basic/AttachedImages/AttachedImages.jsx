import PropTypes from 'prop-types';
import React from 'react';
import { deleteIcon, singleImage, image, wrapper } from './AttachedImages.scss';
import Icon from '../Icon';

const AttachedImages = ({ images, onDelete, className, expandImage }) => (
  <div className={wrapper}>
    {images.map((item, i) => (
      <div key={item.id} className={`${singleImage} ${className}`} >
        <Icon name="close" className={deleteIcon} onClick={() => onDelete(item.id)} />
        <img className={image} src={item.url} alt="" onClick={() => expandImage(images, i)} />
      </div>
    ))}
  </div>
);

export default AttachedImages;

AttachedImages.defaultProps = {
  className: '',
};

AttachedImages.propTypes = {
  images: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired,
  expandImage: PropTypes.func.isRequired,
  className: PropTypes.string,
};
