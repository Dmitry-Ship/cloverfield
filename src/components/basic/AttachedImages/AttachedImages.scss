.singleImage {
  position: relative;
  &:hover > .deleteIcon {
    display: block;
  }
}

.deleteIcon {
  display: none;
  background: rgba(black, .4);
  border-radius: 5px;
  color: rgba(white, .4);
  cursor: pointer;
  position: absolute;
  right: 5%;
  top: 5%;
  &:hover {
    color: white;
  }
}

.image {
  cursor: pointer;
  display: block;
  width: 100%;
}


