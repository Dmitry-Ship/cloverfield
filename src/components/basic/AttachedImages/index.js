export { default } from './AttachedImages.jsx';
