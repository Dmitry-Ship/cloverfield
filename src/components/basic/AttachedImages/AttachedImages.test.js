import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AttachedImages from './AttachedImages';

describe('<AttachedImages />', () => {
  const onDelete = jest.fn();
  const expandImage = jest.fn();
  const images = [
    { id: '1', url: 'image.jpg' },
    { id: '2', url: 'photo.jpg' },
  ];

  it('should render correctly', () => {
    const tree = renderer.create(
      <AttachedImages images={images} expandImage={expandImage} onDelete={onDelete} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onDelete', () => {
    const wrapper = shallow(
      <AttachedImages images={images} expandImage={expandImage} onDelete={onDelete} />
    );

    wrapper.find('Icon').first().simulate('click');

    expect(onDelete).toBeCalledWith(images[0].id);
  });

  it('should call expandImage', () => {
    const wrapper = shallow(
      <AttachedImages images={images} expandImage={expandImage} onDelete={onDelete} />
    );

    wrapper.find('img').first().simulate('click');

    expect(expandImage).toBeCalledWith(images, 0);
  });
});
