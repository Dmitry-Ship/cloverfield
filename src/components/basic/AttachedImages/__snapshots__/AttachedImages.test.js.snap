// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttachedImages /> should render correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="singleImage "
  >
    <i
      className="deleteIcon material-icons"
      onClick={[Function]}
    >
      close
    </i>
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="image.jpg"
    />
  </div>
  <div
    className="singleImage "
  >
    <i
      className="deleteIcon material-icons"
      onClick={[Function]}
    >
      close
    </i>
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="photo.jpg"
    />
  </div>
</div>
`;
